### DISCUSSION TITLE
custom block page for AdGuard Home

### DISCUSSION THREAD

**Discussion by AM-ElZaher** [AdguardTeam/AdGuardHome#5411] [Ideas]
**I Just installed adguard home & use it as my DNS server for my home network and found a workaround to create custom block page as it doesn’t support it by default, here is how I did it**:

•	I have a domain name for my home lab -lets call it **mydomain.com**-
•	My domain name servers are changed to Cloudflare -**optional**-
•	I installed Adguard Home on AWS instance, very basic installation nothing different, I created static ip for it so it won’t change after
reboot and for later use
•	I created new Linux ec2 instance -**let’s call it block**- and installed Apache web server on it,
I also created static ip for it.
•	I created a DNS A record on Cloudflare named it (**blocked.mydomain.com**) and pointed it to my block instance public ip and proxied it.
•	On the block instance I created a simple html page that tell the user that your access is denied because this page is blocked
•	On the adguard home settings >> DNS settings >> blocking mode, I choose custom ip and added the public ip of the block instance.

Now whenever someone tries to access any blocked website the adguard home DNS will redirect the website to my block instance public ip which will show the blocked page html we created earlier.

**Comment by iptvcld** [comment]
does this work via mobile device without getting a cert error asking you to proceed anyways?

**Reply by AM-ElZaher** [reply]
i've installed a cert using let's encrypt to get around this issue so i don't have this error anymore

**Reply by iptvcld** [reply]
to confirm, you installed lets encrypt on the block container (instance)? Are you using nginx? I was wondering you can help perhaps provide your config for that so I can see how your doing it? I have been trying to get this working for a bit now and it works fine on a computer but on a mobile device, i first get a page saying click here to proceed, then it shows the custom block page.

**Reply by AM-ElZaher** [reply]
I created a vm, installed Apache server with my custom block page, certbot and linked this server with my subdomain and got free SSL cert, everything is going just fine, no need to click on click here to proceed or anything on mobile or computer

**Reply by iptvcld** [reply]
Ok so, my setup is:

- I host my adguardHome internally
- I have Ubuntu VM with nginx server with my custom page
- I am using my HAProxy (reverse proxy cert that i already have as a wild card cert *.example.com)
- my domain name would be block.example.com (not registered on CF, i have it resolving internally only
- my nginx server is listening for the ubnunto IP and then is redirecting to block.example.com
- this works on a PC but on mobile it is asking to click on proceed before it shows the block page

Do I need to setup a CNAME on CloudFlare and use its own cert within the VM and just redirect to the html file?

**Reply by iptvcld** [reply]
Sorry; i just configured things like you did and my Apache is still giving the message click to proceed as AGH only allows the use of an IP address and you cannot set an SSL for an IP.

**Reply by iptvcld** [reply]
Would you be able to share your conf?

**Reply by AM-ElZaher** [reply]
Sorry for the late reply, if your internal domain ends with xxx.lan or xxx.local change it to something else like .mynet
Of course add A record on your ADGH to point the ip address to your block domain and then add cname to your address 
I.e : your block page vm is on 10.0.0.5 and will have the domain block.homelab.mynet
So in ADGH add 2 custom rewrites:
block.ip = 10.0.0.5
block.homelab.mynet = block.ip

Final question Is it redirecting to your domain directly or just the ip address on your pc?

**Reply by iptvcld** [reply]
and are you using any certs on your devices?

**Reply by iptvcld** [reply]
ok; i have added the 2 rewrites
block.ip = 192.16.2.246
blocker.example.com = block.ip

in ADGH i have the custom block IP set to 192.168.2.246

But what is doing the redirect from that IP to a secure SSL hostname? Some config needs to listen for the IP on the server and to redirect to the custom hostname

**Comment by AM-ElZaher** [comment]
I assigned a self signed cert with let’s encrypt for the block page vm only

**Reply by iptvcld** [reply]
@AM-ElZaher here i am back at it again trying to make this work..
I installed AGH and my apache server on the same VM
For testing, i have opened port 80 and 443 on this VM and i have go into my cloudflare account added a new A record Block.domain.com to the public IP of my block/AGH VM.

I ended up also adding some configs in the apache server in order for it to redirect http to https and but issue is, some blocked sites it works nice and shows the blocked SSL site and for some blocked sites, it just shows your connection is not private.  I am not sure what I am doing wrong here. Any help would be amazing!!

**Comment by ghostersk** [comment]
Hello @AM-ElZaher I have just sorted this issue with this: https://github.com/ghostersk/catchall-webbrowsing/tree/main
it runs as docker container, and it also records any request on that container. It is not perfect but it helps identify websites when adguard blocks them

**Comment by madcar86** [comment]
late to the party. Is there a way to use the blocked page to enter a admin password to temporarily whitelist the site or even update the whitelist to allow for that device or the whole home?

**Reply by AM-ElZaher** [reply]
i didn't try that, but you can try to play around with authelia to see if it's possible, or even create a custom button to request from the admin temporary access
